sources/nodeUtils.ts

```diff
@@ -4,6 +4,14 @@ export interface NodeError extends Error {
   code: string;
 }
 
+export function isNodeError(err: any): err is NodeError {
+  return `code` in err;
```

Comment:
That's unsafe, we'll get `Cannot use 'in' operator to search for 'code' in undefined` if `err` is nullish
```suggestion
  return !!err?.code;
```